File: ruby/lib/steam/packets/a2a_info_goldsrc_response_packet.rb
autoload "A2A_INFO_ResponsePacket", "steam/packets/a2a_info_response_packet"

# The A2A_INFO_GoldSrcResponsePacket class represents the response to a A2A_INFO
# request send to a GoldSrc server.
class A2A_INFO_GoldSrcResponsePacket < A2A_INFO_ResponsePacket
  
  # Creates a A2A_INFO response object based on the data received.
  def initialize(data)
    super SteamPacket::A2A_INFO_GOLDSRC_RESPONSE_HEADER, data

    @game_ip = @content_data.get_string
    @server_name = @content_data.get_string
    @map_name = @content_data.get_string
    @game_directory = @content_data.get_string
    @game_description = @content_data.get_string
    @number_of_players = @content_data.get_byte
    @max_players = @content_data.get_byte
    @network_version = @content_data.get_byte
    @dedicated = @content_data.get_byte.chr
    @operating_system = @content_data.get_byte.chr
    @password_needed = @content_data.get_byte == 1
    @is_mod = @content_data.get_byte == 1
    
    if @is_mod
      @mod_info = Hash.new
      @mod_info["url_info"] = @content_data.get_string
      @mod_info["url_dl"] = @content_data.get_string
      @content_data.get_byte
      @mod_info["mod_version"] = @content_data.get_long
      @mod_info["mod_size"] = @content_data.get_long
      @mod_info["sv_only"] = @content_data.get_byte == 1
      @mod_info["cl_dll"] = @content_data.get_byte == 1
    end
    
    @secure = @content_data.get_byte == 1
    @number_of_bots = @content_data.get_byte
  end
  
end

File: php/lib/steam/packets/A2A_INFO_GoldSrcResponsePacket.php
<?php
/**
 * @author Sebastian Staudt
 * @license http://www.opensource.org/licenses/bsd-license.php New BSD License
 * @package Steam Condenser (PHP)
 * @subpackage SteamPacket
 * @version $Id$
 */

/**
 * @package Steam Condenser (PHP)
 * @subpackage SteamPacket
 */
class A2A_INFO_GoldSrcResponsePacket extends A2A_INFO_ResponsePacket
{
	/**	
	 * 
	 */
	public function __construct($data)
	{
		parent::__construct(SteamPacket::A2A_INFO_GOLDSRC_RESPONSE_HEADER, $data);
		
		$this->serverIp = $this->contentData->getString();
    $this->serverName = $this->contentData->getString();
    $this->mapName = $this->contentData->getString();
    $this->gameDir = $this->contentData->getString();
    $this->gameDescription = $this->contentData->getString();
    $this->numberOfPlayers = $this->contentData->getByte();
    $this->maxPlayers = $this->contentData->getByte();
    $this->networkVersion = $this->contentData->getByte();
    $this->dedicated = $this->contentData->getByte();
    $this->operatingSystem = $this->contentData->getByte();
    $this->passwordProtected = $this->contentData->getByte() == 1;
    $this->isMod = $this->contentData->getByte() == 1;
    
    if($this->isMod)
    {
      $this->modInfo["urlInfo"] = $this->contentData->getString();
      $this->modInfo["urlDl"] = $this->contentData->getString();
      $this->contentData->getByte();
      $this->modInfo["modVersion"] = $this->contentData->getLong();
      $this->modInfo["modSize"] = $this->contentData->getLong();
      $this->modInfo["svOnly"] = ($this->contentData->getByte() == 1);
      $this->modInfo["clDll"] = ($this->contentData->getByte() == 1);
    }
    
    $this->secure = $this->contentData->getByte() == 1;
    $this->numberOfBots = $this->contentData->getByte();
	}
}
?>
